op-node/rollup/types.go

```diff
@@ -205,11 +205,21 @@ func (cfg *Config) CheckL2ChainID(ctx context.Context, client L2Client) error {
 
 // CheckL2GenesisBlockHash checks that the configured L2 genesis block hash is valid for the given client.
 func (cfg *Config) CheckL2GenesisBlockHash(ctx context.Context, client L2Client) error {
-	l2GenesisBlockRef, err := client.L2BlockRefByNumber(ctx, cfg.Genesis.L2.Number)
+	l2Number := cfg.Genesis.L2.Number
+	l2Hash := cfg.Genesis.L2.Hash
+	// Use the actual genesis on transitioned networks (i.e. op-mainnet & op-goerli).
+	if cfg.L2ChainID.Uint64() == 10 {
+		l2Number = 0
+		l2Hash = common.HexToHash("0x7ca38a1916c42007829c55e69d3e9a73265554b586a499015373241b8a3fa48b")
+	} else if cfg.L2ChainID.Uint64() == 420 {
+		l2Number = 0
+		l2Hash = common.HexToHash("0xc1fc15cd51159b1f1e5cbc4b82e85c1447ddfa33c52cf1d98d14fba0d6354be1")
+	}
```

Comment:
There are chainID constants in this package you could use instead: `opMainnet` and `opGoerli`